### ISSUE TITLE
Accessible error reporting on all forms (admin area and main site)

### ISSUE THREAD

**Issue by denny** [denny/ShinyCMS-ruby#451]
Currently forms only report success/failure, with no clear instructions in the case of failure on what exactly the problem is and how to fix it.

Fields with errors are identified by a red border in the admin area, but this is not an accessible solution; need to add aria-invalid, aria-error, etc.

**Comment by denny** [comment]
If you are a less experienced developer who is looking at this issue and thinking you'd like to try to take it on but you're not entirely confident, I would be happy to help you - whether that's help getting set up, more explanation of what needs doing, pairing on the code task itself, or anything else I can do to help.

In my day job I'm a lead developer who often mentors less experienced developers, and I'd love to use some of that experience to help people take their first steps into contributing to open source projects.

**Comment by MrBowmanXD** [comment]
Seems a good issue to tackle, may i have some explanation about the issue please?

**Comment by denny** [comment]
Hi, thanks for chasing this up 🙂 

As I remember it, most or all of the forms in the admin area, and I think any forms generated using the `ShinyForms` plugin, will let the end-user know if a submitted form has errors - for example, required fields not filled in - but they do that by drawing a red box around the input with the error (for example, a required input left blank), which obviously doesn't help blind people understand what has gone wrong.

There are some bits of accessibility markup that could be added to the forms and form fields to improve this. I think of them as "that aria-* stuff" which shows you about the full extent of my expertise on the subject 😆 

Here's the W3C pages about it: https://www.w3.org/WAI/standards-guidelines/aria/

I think a good starting point would be to try to add aria-* markup to one compulsory field in the admin area - for example, the title box when posting a new blog post - and ask me any questions you run into doing that. Then you can see if the knowledge learned doing that allows you to add accessibility support to most of the other fields in the admin area. I would look at the `ShinyForms` stuff last, once you're feeling more confident about it all, as there's an extra layer of complexity in doing that because those forms are dynamically generated.

How does all that sound? Does it make sense for you as a starting place, or do you have any more specific questions before you can start?
